### PR TITLE
Add support for nil pointer return values

### PR DESCRIPTION
**Pull request by solidDoWant**
This adds support for returning nil pointers from mocked functions. Currently, if the returned value is nil then the cast to the pointer type will cause a panic.

Here's an example:

Previous:
```go
func (c *MockFilesClient) CopyFiles(ctx context.Context, in *CopyFilesRequest, opts ...grpc.CallOption) (*CopyFilesResponse, error) {
	opts0 := []interface{}{ctx, in}
	for _, opts1 := range opts {
		opts0 = append(opts0, opts1)
	}
	args := c.Called(opts0...)
	return args.Get(0), args.Error(1)
}
```

New:
```go
func (c *MockFilesClient) CopyFiles(ctx context.Context, in *CopyFilesRequest, opts ...grpc.CallOption) (*CopyFilesResponse, error) {
	opts0 := []interface{}{ctx, in}
	for _, opts1 := range opts {
		opts0 = append(opts0, opts1)
	}
	args := c.Called(opts0...)
	var ret0 *CopyFilesResponse
	if args.Get(0) != nil {
		ret0 = args.Get(0).(*CopyFilesResponse)
	}
	return ret0, args.Error(1)
}
```

I also ran the makefile target to update the examples, which produced a lot of changes because they're out of date with current protoc tool versions. Let me know if you want me to revert these.

### CODE REVIEW

**Review by felix-kaestner — COMMENTED** [review]
@solidDoWant Nice addition. Keep in mind that using `nil` return values has already been supported through something like

```go
var res *HelloReply
m.OnSayHello(ctx, req).Return(res, nil)
```

or even

```go
m.OnSayHello(ctx, req).Return((*HelloReply)(nil), nil)
```

However I think this provides a nicer interface to most users. As a small improvement, could we do something like the following to avoid some of the branching?

```diff
diff --git a/internal/framework/testify.go b/internal/framework/testify.go
index d3d0d47..7f31c80 100644
--- a/internal/framework/testify.go
+++ b/internal/framework/testify.go
@@ -276,6 +276,9 @@ func (tm *testifyMocker) generateMethodDefinitions(g *protogen.GeneratedFile, me
 				g.P("if args.Get(", i, ") != nil {")
 				g.P("ret", i, " = args.Get(", i, ").(", r, ")")
 				g.P("}")
+
+				ret[i] = fmt.Sprintf("ret%d", i)
+				continue
 			}
 
 			switch r {
@@ -288,11 +291,7 @@ func (tm *testifyMocker) generateMethodDefinitions(g *protogen.GeneratedFile, me
 			case "error":
 				ret[i] = fmt.Sprintf("args.Error(%d)", i)
 			default:
-				if r.IsPointer() {
-					ret[i] = fmt.Sprintf("ret%d", i)
-				} else {
-					ret[i] = fmt.Sprintf("args.Get(%d).(%v)", i, r)
-				}
+				ret[i] = fmt.Sprintf("args.Get(%d).(%v)", i, r)
 			}
 		}
 		g.P("return ", strings.Join(ret, ", "))

```

And ideally also a test would be great.

**Comment by solidDoWant** [comment]
> And ideally also a test would be great.

Any suggestions on where? I don't see existing tests that I could easily add a test case for.

**Comment by felix-kaestner** [comment]
> Any suggestions on where? I don't see existing tests that I could easily add a test case for.

Would be great to add it inside of the examples in https://github.com/lovoo/protoc-gen-go-grpcmock/blob/main/examples/helloworld/testify/helloworld_test.go. I'd like to use that as sort of a playground to test out any changes we make to this project.

So essentially just adding another test that uses nil as return value, which would panic in the current implementation and will work fine after the additions of this PR.
